packages/next/types/index.d.ts

```diff
@@ -115,9 +118,10 @@ export type GetStaticPropsResult<P> =
 
 export type GetStaticProps<
   P extends { [key: string]: any } = { [key: string]: any },
-  Q extends ParsedUrlQuery = ParsedUrlQuery
+  Q extends ParsedUrlQuery = ParsedUrlQuery,
+  S extends PreviewData = PreviewData
```

Comment:
Can we use a consistent name for this, perhaps `D` instead of P or S?

Reply:
Just pushed the changes :)